Sources/Basics/HTTPClient.swift

```diff
@@ -53,6 +59,14 @@ public struct HTTPClient {
         self.configuration = configuration
         // FIXME: inject platform specific implementation here
         self.underlying = handler ?? URLSessionHTTPClient().execute
+
+        // this queue and semaphore is used to limit the amount of concurrent http requests taking place
+        // the default max number of request chosen to match Concurrency.maxOperations which is the number of active CPUs
+        let maxConcurrentRequests = configuration.maxConcurrentRequests ?? Concurrency.maxOperations
+        self.requestsQueue = OperationQueue()
+        self.requestsQueue.name = "org.swift.swiftpm.http-client"
+        self.requestsQueue.maxConcurrentOperationCount = maxConcurrentRequests
+        self.concurrencySemaphore = DispatchSemaphore(value: maxConcurrentRequests)
```

Comment:
What is the reason the semaphore is needed If the concurrency queue's max operation count is already set?

Reply:
the underlying network operation is async, so the operation queue is not enough and the semaphore is used to park the operation

Reply:
exactly